include/envoy/http/codec.h

```diff
@@ -189,6 +210,9 @@ struct Http2Settings {
   // our default connection-level window also equals to our stream-level
   static const uint32_t DEFAULT_INITIAL_CONNECTION_WINDOW_SIZE = 256 * 1024 * 1024;
   static const uint32_t MAX_INITIAL_CONNECTION_WINDOW_SIZE = (1U << 31) - 1;
+
+  // By default, do not enforce stream buffer limits beyond the connection level limits.
+  static const uint32_t DEFAULT_PER_STREAM_BUFFER_LIMIT = 0;
```

Comment:
This is somewhat related to our offline convos, but I'm wondering if by default we should set stream limits (since defaults are huge, maybe stream limit could default to 128MiB or less vs. 256MiB)? Also,  from a config perspective, what is the reasoning behind differentiating between per_stream_buffer_limit_ and initial_stream_window_size_? (I will look more through but figured I would quickly as this question since I noticed it).